src/windows/gfx/d3d11.c

```diff
@@ -278,11 +278,30 @@ static HRESULT d3d11_reload_textures(struct d3d11 *ctx, ID3D11Device *device, ID
 	switch (desc->format) {
 		case MTY_COLOR_FORMAT_BGRA:
 		case MTY_COLOR_FORMAT_AYUV:
+		case MTY_COLOR_FORMAT_Y410:
 		case MTY_COLOR_FORMAT_BGR565:
-		case MTY_COLOR_FORMAT_BGRA5551: {
-			DXGI_FORMAT format = desc->format == MTY_COLOR_FORMAT_BGR565 ? DXGI_FORMAT_B5G6R5_UNORM :
-				desc->format == MTY_COLOR_FORMAT_BGRA5551 ? DXGI_FORMAT_B5G5R5A1_UNORM : DXGI_FORMAT_B8G8R8A8_UNORM;
-			uint8_t bpp = (desc->format == MTY_COLOR_FORMAT_BGRA || desc->format == MTY_COLOR_FORMAT_AYUV) ? 4 : 2;
+		case MTY_COLOR_FORMAT_BGRA5551:
+		case MTY_COLOR_FORMAT_RGB10A2:
+		case MTY_COLOR_FORMAT_RGBA16F: {
+			DXGI_FORMAT format = DXGI_FORMAT_B8G8R8A8_UNORM;
+			switch (desc->format) {
+				case MTY_COLOR_FORMAT_BGR565:
+					format = DXGI_FORMAT_B5G6R5_UNORM;
+					break;
+				case MTY_COLOR_FORMAT_BGRA5551:
+					format = DXGI_FORMAT_B5G5R5A1_UNORM;
+					break;
+				case MTY_COLOR_FORMAT_Y410: // according to MSDN, the view format of Y410 is RGB10A2, just like how that of AYUV is BGRA8
+				case MTY_COLOR_FORMAT_RGB10A2:
+					format = DXGI_FORMAT_R10G10B10A2_UNORM;
+					break;
+				case MTY_COLOR_FORMAT_RGBA16F:
+					format = DXGI_FORMAT_R16G16B16A16_FLOAT;
+					break;
+			}
```

Comment:
Missing `MTY_COLOR_FORMAT_BGRA`, just put this in a separate static function above this, and give dxgi_format, and bpp as out arguments.